v22.0.1 - Windows: kcadm.bat gives Java Exception

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/cli

### Describe the bug

I have installed v22.0.1 and when trying to run the kcadm.bat to import realms (and other items) the batch file will not run. Even running it by itself (to display the help) will give a Java Exception.

PS C:\Development\Test\manual_installs\keycloak-nightly\bin> .\kcadm.bat
Exception in thread "main" java.lang.NoClassDefFoundError: org/fusesource/jansi/WindowsAnsiOutputStream
        at org.jboss.aesh.console.settings.SettingsImpl.getTerminal(SettingsImpl.java:381)
        at org.jboss.aesh.console.Console.init(Console.java:190)
        at org.jboss.aesh.console.Console.<init>(Console.java:118)
        at org.jboss.aesh.console.AeshConsoleImpl.<init>(AeshConsoleImpl.java:98)
        at org.jboss.aesh.console.AeshConsoleBuilder.create(AeshConsoleBuilder.java:160)
        at org.keycloak.client.admin.cli.KcAdmMain.main(KcAdmMain.java:74)
Caused by: java.lang.ClassNotFoundException: org.fusesource.jansi.WindowsAnsiOutputStream
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        ... 6 more

Java Version: 
openjdk 17.0.8 2023-07-18 LTS
OpenJDK Runtime Environment Zulu17.44+15-CA (build 17.0.8+7-LTS)
OpenJDK 64-Bit Server VM Zulu17.44+15-CA (build 17.0.8+7-LTS, mixed mode, sharing)

Windows Version:
Edition	Windows 10 Enterprise
Version	22H2
Installed on	‎6/‎9/‎2022
OS build	19045.3086
Experience	Windows Feature Experience Pack 1000.19041.1000.0

Running on the same machine with the same version of Java using Keycloak 21.1.2 does not show this error.

The server seems to be running fine with this version of Java. I can access the server, login, configure, etc.

### Version

22.0.1

### Expected behavior

The admin cli will run as expected.

### Actual behavior

PS C:\Development\Test\manual_installs\keycloak-nightly\bin> .\kcadm.bat
Exception in thread "main" java.lang.NoClassDefFoundError: org/fusesource/jansi/WindowsAnsiOutputStream
        at org.jboss.aesh.console.settings.SettingsImpl.getTerminal(SettingsImpl.java:381)
        at org.jboss.aesh.console.Console.init(Console.java:190)
        at org.jboss.aesh.console.Console.<init>(Console.java:118)
        at org.jboss.aesh.console.AeshConsoleImpl.<init>(AeshConsoleImpl.java:98)
        at org.jboss.aesh.console.AeshConsoleBuilder.create(AeshConsoleBuilder.java:160)
        at org.keycloak.client.admin.cli.KcAdmMain.main(KcAdmMain.java:74)
Caused by: java.lang.ClassNotFoundException: org.fusesource.jansi.WindowsAnsiOutputStream
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        ... 6 more

### How to Reproduce?

Install v22.0.1 on Windows and go to the bin folder, run "kcadm.bat". 

### Anything else?

_No response_

--- a/integration/client-cli/admin-cli/pom.xml
+++ b/integration/client-cli/admin-cli/pom.xml
@@ -29,10 +29,20 @@
     <name>Keycloak Admin CLI</name>
     <description/>
 
+    <properties>
+        <jansi.version>1.18</jansi.version>
+    </properties>
+
     <dependencies>
         <dependency>
             <groupId>org.jboss.aesh</groupId>
             <artifactId>aesh</artifactId>
+        </dependency>
+        <!-- Jansi library version needs to be overridden due to the backwards compatibility - see #21851 -->
+        <dependency>
+            <groupId>org.fusesource.jansi</groupId>
+            <artifactId>jansi</artifactId>
+            <version>${jansi.version}</version>
         </dependency>
         <dependency>
             <groupId>org.keycloak</groupId>

